ecosystem/node-checker/src/configuration/types.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(dead_code)]
+
+use crate::{metric_evaluator::StateSyncMetricsEvaluatorArgs, runner::BlockingRunnerArgs};
+use anyhow::Result;
+use clap::Parser;
+use once_cell::sync::Lazy;
+use poem_openapi::{types::Example, Object as PoemObject};
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+pub const DEFAULT_METRICS_PORT: u16 = 9101;
+pub const DEFAULT_API_PORT: u16 = 8080;
+pub const DEFAULT_NOISE_PORT: u16 = 6180;
+
+pub static DEFAULT_METRICS_PORT_STR: Lazy<String> =
+    Lazy::new(|| format!("{}", DEFAULT_METRICS_PORT));
+pub static DEFAULT_API_PORT_STR: Lazy<String> = Lazy::new(|| format!("{}", DEFAULT_API_PORT));
+pub static DEFAULT_NOISE_PORT_STR: Lazy<String> = Lazy::new(|| format!("{}", DEFAULT_NOISE_PORT));
+
+// To briefly explain why many of these structs derive 3 different classes of traits:
+// - Parser (clap): To allow users to generate configs easily using nhc configuration create
+// - Serialize / Deserialize (serde): So we can read / write configs from / to disk
+// - PoemoOject: So we can return the configuration over the API
+
+#[derive(Clone, Debug, Deserialize, Parser, PoemObject, Serialize)]
+#[clap(author, version, about, long_about = None)]
+pub struct NodeConfiguration {
+    #[clap(flatten)]
+    pub node_address: NodeAddress,
+
+    /// This is the name we expect clients to send over the wire to select
+    /// which configuration they want to use. e.g. devnet_fullnode
+    #[clap(long)]
+    pub configuration_name: String,
+
+    /// This is the name we will show for this configuration to users.
+    /// For example, if someone opens the NHC frontend, they will see this name
+    /// in a dropdown list of configurations they can test their node against.
+    /// e.g. "Devnet FullNode", "Testnet Validator Node", etc.
+    #[clap(long)]
+    pub configuration_name_pretty: String,
+
+    /// The chain ID we expect to find when we speak to the node.
+    /// If not given, we will just assume the value we find is correct.
+    /// If given, we will check that the value is correct, exiting if not.
+    #[clap(long)]
+    chain_id: Option<u16>,
```

Comment:
My thinking here is this is an unnecessary extra burden on the NHC runner, since we can discover this value easily by hitting the baseline. I intend to do this:
```
for configuration in configurations {
    let (chain_id, role_type) = get_chain_id_and_role_type(configuration.url);
    if let Some(config_chain_id) = configuration.chain_id {
        if config_chain_id != chain_id {
            bail
        }
    }
    // same thing for role_type
}
```
I think this is a nice ergonomic win, particularly for things like devnet, where the chain ID is constantly changing.